### PR TITLE
Add flutterwave env variable

### PR DESCRIPTION
**Pull request by neelxie**
# Description
Add flutter wave environment variable

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Trello Ticket ID
https://trello.com/c/N7Cgo9HV

### CODE REVIEW

File: .circleci/bin/build-and-push-frontend.sh [hunk 1]
```diff
@@ -5,6 +5,7 @@ set -e
 
 # env variables
 REACT_APP_BASE_URL=${REACT_APP_API_BASE_URL:-}
+REACT_APP_PUBLIC_KEY=${REACT_APP_PUBLIC_KEY:-}
```
Comment (inno-asiimwe) [inline .circleci/bin/build-and-push-frontend.sh:8]:
```suggestion
REACT_APP_FLUTTERWAVE_PUBLIC_KEY=${REACT_APP_PUBLIC_KEY:-}
```

Comment (inno-asiimwe) [reply .circleci/bin/build-and-push-frontend.sh:8]: I suggest that we use more descriptive names for the variables. We may have more public keys in future.

Comment (neelxie) [reply .circleci/bin/build-and-push-frontend.sh]: Thank you, I have made changes

**Comment by inno-asiimwe** [comment]
```
function create_env_file
{
    echo REACT_APP_API_BASE_URL=$REACT_APP_API_BASE_URL >> .env

    if [ $branch == 'master' ]
    then
        echo REACT_APP_GITHUB_CLEINT_ID=$REACT_APP_GITHUB_CLEINT_ID_PROD >> .env
    else
        echo REACT_APP_GITHUB_CLEINT_ID=$REACT_APP_GITHUB_CLEINT_ID_STAGING >> .env
    fi
}
```
You will need to update this method as well. It creates an actual `.env` .

**Review by e-ian — DISMISSED** [review]
Please address the comments

**Review by inno-asiimwe — CHANGES_REQUESTED** [review]
https://github.com/crane-cloud/frontend/pull/509#issuecomment-1098134865

**Comment by inno-asiimwe** [comment]
@neelxie Just to be sure, is this the same key that will be used for production?

File: .circleci/bin/build-and-push-frontend.sh [hunk 2]
```diff
@@ -15,6 +16,7 @@ branch=${CIRCLE_BRANCH:-}
 function create_env_file
 {
     echo REACT_APP_API_BASE_URL=$REACT_APP_API_BASE_URL >> .env
+    echo REACT_APP_FLUTTERWAVE_PUBLIC_KEY=$REACT_APP_PUBLIC_KEY >> .env
```
Comment (inno-asiimwe) [inline .circleci/bin/build-and-push-frontend.sh:19]:
```suggestion
    echo REACT_APP_FLUTTERWAVE_PUBLIC_KEY=$REACT_APP_PUBLIC_KEY >> .env
```
```suggestion
    echo REACT_APP_PUBLIC_KEY=$REACT_APP_FLUTTERWAVE_PUBLIC_KEY >> .env
```

[the remark this answers is not in the document]
Comment (inno-asiimwe) [reply .circleci/bin/build-and-push-frontend.sh:19]: The app expects `REACT_APP_PUBLIC_KEY`

[the remark this answers is not in the document]
Comment (inno-asiimwe) [reply .circleci/bin/build-and-push-frontend.sh:19]: The app expects `REACT_APP_PUBLIC_KEY`

File: src/config.js [hunk 3]
```diff
@@ -5,3 +5,4 @@ export const DOCS_URL = "https://docs.cranecloud.io";
 export const BLOG_URL = "https://medium.com/cranecloud";
 //github auth urls
 export const GIT_REDIRECT_URL = `https://github.com/login/oauth/authorize?client_id=${process.env.REACT_APP_GITHUB_CLEINT_ID}&scope=read:user,user:email`;
+export const FLUTTER_WAVE_PUBLIC_KEY = process.env.REACT_APP_FLUTTER_WAVE_PUBLIC_KEY
```
Comment (inno-asiimwe) [inline src/config.js:8]:
```suggestion
export const FLUTTER_WAVE_PUBLIC_KEY = process.env.REACT_APP_FLUTTERWAVE_PUBLIC_KEY
```

**Comment by neelxie** [comment]
> @neelxie Just to be sure, is this the same key that will be used for production?

@inno-asiimwe no this is for staging and testing purposes and I have done the necessary changes, do check them out.

**Review by inno-asiimwe — APPROVED** [review]
Good stuff.

**Pull request merged**